### ISSUE TITLE
Make it possible to install any version from brew

### ISSUE THREAD

**Issue by grzegorzkrukowski** [realm/SwiftLint#5784]
### New Issue Checklist

- [ x ] I've Updated SwiftLint to the latest version.
- [ x ] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Feature or Enhancement Proposal

At the moment you can install only the latest version from Brew, it's tough to lock your SwiftLint version on CI/CD or even locally for development.
It would be great to have a control over the SwiftLint version being installed this way:
`brew install swiftlint@0.55.0`

For example, a similar solution is possible with [postgress](https://formulae.brew.sh/formula/postgresql@12#default).

**Comment by ahmdyasser** [comment]
Hey, @grzegorzkrukowski, do you think this solves the root issue? Because if SwiftLint added other versions for example, how many versions will be available? I'm seeing that Postgres are providing only 7 major releases, which still is problematic.
I want to lock the version I'm using with full control over the minor, and patch number too (1.x.x).

I'd suggested couple of possible solutions:

- [Mint](https://github.com/yonaskolb/Mint): It gives you full control over versions, BUT it only works with SPM (if you have a golang CLI tool, mint can't handle it) example: [Lokalise CLI](https://github.com/lokalise/lokalise-cli-2-go/releases)
- [Executable](https://github.com/realm/SwiftLint/releases/tag/0.57.0): You can download them from the releases section, BUT then you have to download the and update your repo manually.
- [Mise](https://mise.jdx.dev/): It works like a charm, but the spm backend is still experimental. You can also create a custom plugin to avoid the experimental issue.

**Comment by SimplyDanny** [comment]
I would like to add another possible solution:

* Consume SwiftLint as a Command Plugin in SPM. There you can specify exact versions. With the introduction of [SwiftLintPlugins](https://github.com/SimplyDanny/SwiftLintPlugins), this doesn't mean you need to compile the whole SwiftLint with all its dependencies. The download of the binary package is enough. Then it's only a matter of running `swift package plugin swiftlint`.

**Comment by grzegorzkrukowski** [comment]
For now, we decided to go with [SwiftLintPlugins](https://github.com/SimplyDanny/SwiftLintPlugins), which works well for us.
Unfortunately, it requires full dependency resolution before we can use it, and getting the binary path is a bit tricky - but all solved and we will stick to that now.

It is a nice solution that works for both local developers environment and CI.
So I can recommend that approach.

**Comment by ahmdyasser** [comment]
> I would like to add another possible solution:
> 
> * Consume SwiftLint as a Command Plugin in SPM. There you can specify exact versions. With the introduction of [SwiftLintPlugins](https://github.com/SimplyDanny/SwiftLintPlugins), this doesn't mean you need to compile the whole SwiftLint with all its dependencies. The download of the binary package is enough. Then it's only a matter of running `swift package plugin swiftlint`.

Hey @SimplyDanny, thanks for the suggestion. I have a question, if we are talking about binaries here, why not using SwiftLint binaries directly? I mean instead of `swift package plugin swiftlint` we can do `./swiftlint-executable` from the official repo.

**Comment by SimplyDanny** [comment]
> Hey @SimplyDanny, thanks for the suggestion. I have a question, if we are talking about binaries here, why not using SwiftLint binaries directly? I mean instead of `swift package plugin swiftlint` we can do `./swiftlint-executable` from the official repo.

Sure. You can always use the executables from the release page. But this is manual work.

> Also updating it will be manually right? I don't see it as a big problem because we don't update SwiftLint version frequently, but it's something to be aware about.

If you're using SPM anyway, it's as easy as adding the plugin repository as another dependency with the full support of SPM's version syntax. This works out-of-the-box right now locally as well as on CI. I see this as significant advantages compared to using the executables.

**Comment by ahmdyasser** [comment]
> If you're using SPM anyway, it's as easy as adding the plugin repository as another dependency with the full support of SPM's version syntax. This works out-of-the-box right now locally as well as on CI. I see this as significant advantages compared to using the executables.

You mean I can point to a plugin binary repository and from there it will work out of the box as same as dealing with a regular Swift package!! wow! this is really helpful I didn't know that. Thanks a lot.

**Comment by SimplyDanny** [comment]
> You mean I can point to a plugin repository that points to the original tool binaries and from there it will work out of the box as same as dealing with a regular Swift package!! wow! this is really helpful I didn't know that. Thanks a lot.

Yes! You just add

```swift
.package(url: "https://github.com/SimplyDanny/SwiftLintPlugins", exact: "0.57.0")
```

to your `Package.swift` and then you can call SwiftLint via `swift package plugin swiftlint`. The build only compiles the plugin code itself but the whole `swiftlint` binary is downloaded in its pre-compiled version. It doesn't need to resolve dependencies, download and build them.

**Comment by ppamorim** [comment]
I have the same issue, we are unable to downgrade `swiftlint`. The current version `0.58.2` added a false positive and I am trying to downgrade to test it. But I simply can't.

**Comment by SimplyDanny** [comment]
> I have the same issue, we are unable to downgrade `swiftlint`. The current version `0.58.2` added a false positive and I am trying to downgrade to test it. But I simply can't.

Please report the false positive as well.
